### PR TITLE
In our organization we have some teams. All teams works in the same G…

### PR DESCRIPTION
**Pull request by fkieling**
It's a nice project! I like it.
In our organization we have some teams. All teams works in the same GitHub organization and use a prefix on their repositories. With this change our teams will get their own check on icinga for their own repositories.

### CODE REVIEW

File: security-alert-notifier.rb [hunk 1]
```diff
@@ -177,17 +181,33 @@ def fetch_repositories(http:, cursor: nil)
     github = GitHub.new
 
     if github.vulnerable_repos.any?
-      total_vulnerabilities = github.vulnerable_repos.sum { |repo| repo.alerts.length }
-      puts "WARNING: #{total_vulnerabilities} vulnerabilities in #{github.vulnerable_repos.length} repos"
+      if options[:repos].nil?
+        total_vulnerabilities = github.vulnerable_repos.sum { |repo| repo.alerts.length }
+        vulnerable_repos = github.vulnerable_repos.length
```
Comment (erbridge) [inline security-alert-notifier.rb:186]: Think we could rename this variable to be a bit clearer that it's a count of repos, not the repos themselves (especially given `github.vulnerable_repos` has the same name)?

**Review by erbridge — CHANGES_REQUESTED** [review]
This looks like a useful feature. Thanks for the contribution.

I have a small variable naming request.

I'm also on the fence about whether `repos` is the right name for the option 🤔. My inital reaction was that it should be `filter`, but I'm not sure. Open to thoughts.

**Comment by erbridge** [comment]
Please could you add test coverage for this change?

File: security-alert-notifier.rb (hunk 1 above)
Comment (fkieling) [reply security-alert-notifier.rb:186]: Ok. `repos` for this option is irritiating. Your first inspiration is good. Let's call it `filter`

You're right. `vulnerable_repo` isn't so good. `vulnerable_repo_count` puts it better.
```suggestion
        vulnerable_repo_count = github.vulnerable_repos.length
```

File: security-alert-notifier.rb [hunk 2]
```diff
@@ -177,17 +181,33 @@ def fetch_repositories(http:, cursor: nil)
...
+        total_vulnerabilities = github.vulnerable_repos.sum { |repo| repo.alerts.length }
+        vulnerable_repo_count = github.vulnerable_repos.length
+      else
+        total_vulnerabilities = 0
+        vulnerable_repos = 0
+        github.vulnerable_repos.each do |repo|
+          if options[:repos].nil? || repo.url =~ /#{options[:repos]}/
+            vulnerable_repos = vulnerable_repos + 1
+            repo.alerts.each do |alert|
+              total_vulnerabilities = total_vulnerabilities + 1
```
Comment (fkieling) [inline security-alert-notifier.rb:194]:
```suggestion
              vulnerable_repo_count = vulnerable_repo_count + 1
```
rename for better voriable name

File: security-alert-notifier.rb [hunk 3]
```diff
@@ -177,17 +181,33 @@ def fetch_repositories(http:, cursor: nil)
...
+            end
+          end
+        end
+      end
+      puts "WARNING: #{total_vulnerabilities} vulnerabilities in #{vulnerable_repos} repos"
```
Comment (fkieling) [inline security-alert-notifier.rb:199]:
```suggestion
      puts "WARNING: #{total_vulnerabilities} vulnerabilities in #{vulnerable_repo_count} repos"
```
rename for better voriable name

**Comment by fkieling** [comment]
How can i add test coverage for something that isn't in github class? Must i mock http? Or is'nt necessary to test this change?

**Comment by erbridge** [comment]
You're right, sorry! Looking good. I'll get it merged.

**Comment by erbridge** [comment]
Merged in 976caa0. Thanks for your contribution!

**Pull request closed without merging**